service/api/cadence/client.go

```diff
@@ -113,6 +114,28 @@ func (t *cadenceClient) StartInterpreterWorkflow(ctx context.Context, options ap
 	return run.RunID, nil
 }
 
+func (t *cadenceClient) StartWaitForStateCompletionWorkflow(ctx context.Context, options api.StartWorkflowOptions) (runId string, err error) {
+	workflowOptions := client.StartWorkflowOptions{
+		ID:                           options.ID,
+		TaskList:                     options.TaskQueue,
+		WorkflowIDReusePolicy:        client.WorkflowIDReusePolicyRejectDuplicate,
+		ExecutionStartToCloseTimeout: 600 * time.Second, // TODO, make this configurable
+	}
+	run, err := t.cClient.StartWorkflow(ctx, workflowOptions, cadence.WaitforStateCompletionWorkflow)
+	if err != nil {
+		if t.IsWorkflowAlreadyStartedError(err) {
+			describeResponse, error := t.cClient.DescribeWorkflowExecution(ctx, workflowOptions.ID, "")
```

Comment:
the already started error contains the runId. Also within this file, there is no need to use `IsWorkflowAlreadyStartedError`

Reply:
actually we probably don't need to return runId?

Reply:
is runId a required param in GetWorkflowResult?